src/app/forms/services/dynamic-form.types.ts

```diff
@@ -176,7 +179,9 @@ const cleanValue = (form: CustomFormGroup | CustomFormArray): { [key: string]: a
     } else if (control instanceof CustomFormArray) {
       cleanValue[key] = control.getCleanValue(control);
     } else if (control instanceof CustomFormControl) {
-      if (control.meta.type === FormNodeTypes.CONTROL && !control.meta.hide) {
+      if (control.meta.type === FormNodeTypes.CONTROL && control.meta.required && control.meta.hide) {
+        Array.isArray(cleanValue) ? cleanValue.push(control.meta.value || null) : (cleanValue[key] = control.meta.value || null);
```

Comment:
Why bother with `|| null`? How does passing `undefined` or `null` differ from each other here?

Reply:
when parsing form values as `json`, `undefined` values omit the property while `null` is a value.

Reply:
Ah, I see. Fair enough 👍